ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule.hs

```diff
@@ -586,14 +589,25 @@ newLongRangeAttack ::
 newLongRangeAttack g BlockTree {btTrunk, btBranches = [branch]} = do
   honest <- peerScheduleFromTipPoints g honParams [(IsTrunk, [AF.length btTrunk - 1])] btTrunk []
   adv <- peerScheduleFromTipPoints g advParams [(IsBranch, [AF.length (btbFull branch) - 1])] btTrunk [btbFull branch]
-  pure (fromSchedulePoints (Map.fromList [(HonestPeer, honest), ("adversary 1", adv)]))
+  pure (fromSchedulePoints (mkPeers honest [adv]))
   where
     honParams = defaultPeerScheduleParams {pspHeaderDelayInterval = (0.3, 0.4)}
     advParams = defaultPeerScheduleParams {pspTipDelayInterval = (0, 0.1)}
 
 newLongRangeAttack _ _ =
   pure Nothing
 
+uniform ::
+  StatefulGen g m =>
+  g ->
+  BlockTree TestBlock ->
+  m (Maybe PointSchedule)
+uniform g BlockTree {btTrunk, btBranches} = do
+  honest <- peerScheduleFromTipPoints g defaultPeerScheduleParams [(IsTrunk, [AF.length btTrunk - 1])] btTrunk []
+  advs <- for btBranches $ \ branch ->
+    peerScheduleFromTipPoints g defaultPeerScheduleParams [(IsBranch, [AF.length (btbFull branch) - 1])] btTrunk [btbFull branch]
+  pure (fromSchedulePoints (mkPeers honest advs))
```

Comment:
Some tweaks to improve coverage:

* have one or more peers do rollbacks
* allow the node under test to catch up in the middle of the test instead of only at the end (tip points would have the shape of `singleJumpTipPoints` and `rollbacksTipPoints`)
* randomize a bit the generation parameters used for the different peers